catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -0,0 +1,249 @@
+
+/** Provides an internal REST interface for working with custom form data for Intrigue. */
+public class SearchFormsApplication implements SparkApplication {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SearchFormsApplication.class);
+
+  private static final Security SECURITY = Security.getInstance();
+
+  private static final ObjectMapper MAPPER =
+      JsonFactory.create(
+          new JsonParserFactory().usePropertyOnly(),
+          new JsonSerializerFactory()
+              .addPropertySerializer(new FilterNodeValueSerializer())
+              .useAnnotations()
+              .includeEmpty()
+              .includeDefaultValues()
+              .setJsonFormatForDates(false));
+
+  private final CatalogFramework catalogFramework;
+
+  private final EndpointUtil util;
+
+  public SearchFormsApplication(CatalogFramework catalogFramework, EndpointUtil util) {
+    this.catalogFramework = catalogFramework;
+    this.util = util;
+  }
+
+  /**
+   * Called via blueprint on initialization. Reads configuration in {@code etc/forms} and
+   * initializes Solr with query and result templates using the {@link Metacard#TITLE} field as a
+   * unique key.
+   */
+  public void setup() {
+    Function<Map<String, Result>, Set<String>> titles =
+        map ->
+            map.entrySet()
+                .stream()
+                .map(Map.Entry::getValue)
+                .map(Result::getMetacard)
+                .map(Metacard::getTitle)
+                .collect(Collectors.toSet());
+
+    Set<String> queryTitles = queryAsAdmin(FormAttributes.Query.TAG, titles);
+    Set<String> resultTitles = queryAsAdmin(FormAttributes.Result.TAG, titles);
+
+    List<Metacard> systemTemplates = config().get();
+    if (systemTemplates.isEmpty()) {
+      return;
+    }
+
+    Stream<Metacard> dedupedTemplateMetacardStream =
+        Stream.concat(
+            systemTemplates
+                .stream()
+                .filter(QueryTemplateMetacardImpl::isQueryTemplateMetacard)
+                .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
+            systemTemplates
+                .stream()
+                .filter(ResultTemplateMetacardImpl::isResultTemplateMetacard)
+                .filter(metacard -> !resultTitles.contains(metacard.getTitle())));
+
+    saveMetacards(dedupedTemplateMetacardStream.collect(Collectors.toList()));
+  }
+
+  /** Spark's API-mandated init (not OSGi related) for registering REST functions. */
+  @Override
+  public void init() {
+    get(
+        "/forms/query",
+        (req, res) ->
+            MAPPER.toJson(
+                util.getMetacardsByFilter(FormAttributes.Query.TAG)
+                    .entrySet()
+                    .stream()
+                    .map(Map.Entry::getValue)
+                    .map(Result::getMetacard)
+                    .map(this::toFormTemplate)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList())));
```

Comment:
if you don't want the streaming wrapped inside the mapper function you can always put the mapper function as the optional 3rd parameter for the spark methods. 
```java
get(
        "/forms/query",
        (req, res) ->
            util.getMetacardsByFilter(FormAttributes.Query.TAG)
                .entrySet()
                .stream()
                .map(Map.Entry::getValue)
                .map(Result::getMetacard)
                .map(this::toFormTemplate)
                .filter(Objects::nonNull)
                .collect(Collectors.toList()),
        MAPPER::toJson);
```